src/main/cc/proxy_client/proxy_client.cc

```diff
@@ -658,16 +647,26 @@ int ExecuteCommand(int argc, char** argv, const char** env) {
 
 int SelectAndRunCommand(int argc, char** argv, const char** env) {
   srand(time(nullptr));
-  // Hack to allow goma_ctl ensure_start to work. This is called by
-  // the Android @head build when USE_GOMA is set.
-  if (!strcmp(argv[1], "tmp_dir")) {
-    const char *tmp_dir = "/tmp/goma_tmp";
-    mkdir(tmp_dir, 0777);
-    cout << tmp_dir << "\n";
-    return 0;
+  gflags::SetUsageMessage("RBE client for remote proxy");
+  int sep_idx = argc;
+  for (int i = 1; i < argc; i++) {
+    if (!strcmp(argv[i], "--")) {
+      sep_idx = i;
+      break;
+    }
   }
-
-  if (!strcmp(argv[1], "list_includes")) {
+  gflags::ParseCommandLineFlags(&sep_idx, &argv, false);
+  if (FLAGS_command == "run") {
+      if (sep_idx == argc) {
+        cout << "You set the command to 'run'.\n";
+         cout << "Usage: " << gflags::ProgramInvocationShortName() <<
+              " [--arg=val] --command=run -- [full command]\n";
+         return 1;
+      }
+      return ExecuteCommand(argc-sep_idx-1, &argv[sep_idx+1], env);
+  } else if (FLAGS_command == "include_stats") {
```

Comment:
No need for else, every if has a return.

Reply:
Done